packages/frontend/src/components/ControlledAutocomplete/AddNewOptionModal.tsx

```diff
@@ -0,0 +1,125 @@
+import { FormEvent, useCallback, useState } from 'react'
+import { useMutation } from '@apollo/client'
+import {
+  FormControl,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+} from '@chakra-ui/react'
+import { Button, FormLabel, Input } from '@opengovsg/design-system-react'
+import client from 'graphql/client'
+import { CREATE_TABLE } from 'graphql/mutations/tiles/create-table'
+import { GET_DYNAMIC_DATA } from 'graphql/queries/get-dynamic-data'
+
+interface AddNewOptionalModalProps {
+  modalHeader: string
+  onSubmit: (newValue: string) => void
+  onClose: () => void
+}
+
+interface UseCreateNewOptionProps {
+  inputValue: string
+  addNewId?: string
+}
+
+export function useCreateNewOption(setValue: (newValue: string) => void) {
+  const [createTable] = useMutation(CREATE_TABLE)
+  const [isCreatingNewOption, setIsCreatingNewOption] = useState(false)
+  const createNewOption = useCallback(
+    async ({ inputValue, addNewId }: UseCreateNewOptionProps) => {
+      if (!inputValue.trim() || !addNewId) {
+        return
+      }
+      let newValue: string | undefined
+      setIsCreatingNewOption(true)
+      try {
+        switch (addNewId) {
+          case 'tiles-createTileRow-tableId': {
+            const { data } = await createTable({
+              variables: {
+                input: {
+                  name: inputValue.trim(),
+                  isBlank: true,
+                },
+              },
+            })
+            newValue = data?.createTable.id
+            break
+          }
+          default:
+            break
+        }
+        await client.refetchQueries({ include: [GET_DYNAMIC_DATA] })
+        if (newValue) {
+          setValue(newValue)
+        }
+      } finally {
+        setIsCreatingNewOption(false)
+      }
+    },
+    [createTable, setValue],
+  )
+  return { createNewOption, isCreatingNewOption }
+}
+
+function AddNewOptionalModal({
+  modalHeader,
+  onClose,
+  onSubmit,
+}: AddNewOptionalModalProps) {
+  const [inputValue, setInputValue] = useState('')
+  const trimmedInputValue = inputValue.trim()
+
+  const onFormSubmit = useCallback(
+    async (event: FormEvent<HTMLFormElement>) => {
+      event.preventDefault()
+      if (!trimmedInputValue) {
+        return
+      }
+      onSubmit(trimmedInputValue)
+    },
+    [onSubmit, trimmedInputValue],
+  )
+
+  return (
+    <Modal
+      isOpen={true}
+      onClose={onClose}
+      size="md"
+      closeOnEsc={false}
+      motionPreset="none"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <form onSubmit={onFormSubmit}>
+          <ModalHeader>{modalHeader}</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody pb={8}>
+            <FormControl display="flex" flexDir="column" gap={2}>
+              {/* This is hardcoded for now, will change when there are more of such actions */}
```

Comment:
nit: ideally we should be able to reuse `CreateTileModal`. Maybe next person to hit this code should do the refactoring

Reply:
out of scope for now. this modal was meant to be a generic one input modal for other add-new dropdowns in future .